File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
import math
def solution(n, words):
    answer = []
    word_list = []
    word_list.append(words[0])
    for i in range(1, len(words)):
        # 말했던 단어이거나 이어서 말하지 못한 경우
        if words[i] in word_list or word_list[-1][-1] != words[i][0]:

            return  [i % n + 1, i // n + 1]
        else:
            word_list.append(words[i])
    return [0,0]    

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let ans = [0, 0];
    let stack = [];
    
    for(let i = 0; i < words.length; i++){
        if(stack.length === 0){
            stack.push(words[i]);
            continue;
        }
        
        const pre = stack[stack.length - 1];
        const preWord = pre[pre.length - 1];
        
        const post = words[i];
        const postWord = post[0];
        
        if(preWord !== postWord || stack.includes(post)){
            const who = (i % n) + 1;
            const count = Math.ceil((i + 1) / n);
            
            ans[0] = who;
            ans[1] = count;
            break;
        }
        
        stack.push(words[i]);
    }
    
    return ans;
}